apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Internal/Logging/Logger.cs

```diff
@@ -13,172 +13,304 @@
 // limitations under the License.
 
 using System;
+using System.Collections;
```

Comment:
I'm going to assume that all the implementation here was just moved from DefaultLogger - I haven't reviewed it in any detail.

Reply:
that's correct